### PR TITLE
(#72) Support upload of artifacts via maven client

### PR DESCRIPTION
**Pull request by victornoel**
This is for #72.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#82](https://github.com/artipie/maven-adapter/issues/82) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/95773f3a-6b7b-4d9c-a44f-d4f9e94bf295); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/maven/MavenHttpITCase.java [hunk 1]
```diff
@@ -79,34 +91,127 @@ void tearDown() {
     }
 
     @Test
-    void serveArtifact(final @TempDir Path localrepo, final @TempDir Path remote)
-        throws Exception {
-        final FileStorage storage = new FileStorage(remote, this.vertx.fileSystem());
-        final Path artifact = remote.resolve("org").resolve("apache").resolve("maven")
-            .resolve("resolver").resolve("maven-resolver-util").resolve("1.3.3");
-        Files.createDirectories(artifact);
+    void serveArtifact(final @TempDir Path temp) throws Exception {
```
Comment (Vatavuk) [inline src/test/java/com/artipie/maven/MavenHttpITCase.java:94]: @victornoel I think this is simpler: `(final @TempDir Path local,  final@TempDir Path remote)`

File: src/test/java/com/artipie/maven/MavenHttpITCase.java [hunk 2]
```diff
@@ -79,34 +91,127 @@ void tearDown() {
...
+                    port,
+                    Files.createDirectories(temp.resolve("local"))
+                ).artifact("org.apache.maven.resolver:maven-resolver-util:1.3.3"),
+                new IsNot<>(new IsNull<>())
+            );
+        }
+    }
+
+    @Test
+    void deployOneArtifact(final @TempDir Path temp)
```
Comment (Vatavuk) [inline src/test/java/com/artipie/maven/MavenHttpITCase.java:117]: @victornoel I think this is simpler: `(final @TempDir Path local,  final@TempDir Path remote)`

File: src/test/java/com/artipie/maven/MavenHttpITCase.java (hunk 1 above)
Comment (victornoel) [reply src/test/java/com/artipie/maven/MavenHttpITCase.java:94]: @Vatavuk unfortunately there is an undocumented bug in junit: if you put multiple `@TempDir` in a test, they will all point to the same temporary directory on the filesystem. See https://github.com/junit-team/junit5/issues/2219 for details. That's why I changed it.

File: src/test/java/com/artipie/maven/MavenHttpITCase.java (hunk 2 above)
Comment (victornoel) [reply src/test/java/com/artipie/maven/MavenHttpITCase.java:117]: @Vatavuk see my comment above

File: src/test/java/com/artipie/maven/MavenHttpITCase.java (hunk 1 above)
Comment (Vatavuk) [reply src/test/java/com/artipie/maven/MavenHttpITCase.java:94]: @victornoel ok

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find at least 3 issues during next CR, thanks

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
> @0crat quality acceptable  ([here](https://github.com/artipie/maven-adapter/issues/82#issuecomment-626932923))

@sereshqua There is no quality review for [#82](https://github.com/artipie/maven-adapter/issues/82), no performer

**Pull request merged**